geopmdpy/geopmdpy/session.py

```diff
@@ -256,15 +287,29 @@ def check_read_args(self, run_time, period):
             raise RuntimeError('Specified a period greater than 24 hours')
         if period < 0.0 or run_time < 0.0:
             raise RuntimeError('Specified a negative run time or period')
+        if report_samples is not None:
+            if report_samples < 0:
+                raise RuntimeError('Specified report samples is negative')
+            if int(report_samples) != report_samples:
+                raise RuntimeError(f'Specified report samples is non-integer: {report_samples}')
```

Comment:
Shouldn't these if checks be reversed since you're doing an integer operation with the ``< 0 check``?
```suggestion
            if int(report_samples) != report_samples:
                raise RuntimeError(f'Specified report samples is non-integer: {report_samples}')
            if report_samples < 0:
                raise RuntimeError('Specified report samples is negative')

```

Reply:
On closer inspection, argparse does the integer check for us. I think the ``if int(report_samples) != report_samples`` is dead code:

```
$ geopmsession -s 1.234
usage: geopmsession [-h] [-v] [-t TIME] [-p PERIOD] [--pid PID] [--print-header | -n] [-d DELIMITER] [-r REPORT_OUT] [-o TRACE_OUT] [--enable-mpi] [-f REPORT_FORMAT] [-s REPORT_SAMPLES]
geopmsession: error: argument -s/--report-samples: invalid int value: '1.234'
```